### PR TITLE
treasury factory contract setup and web4 implementation

### LINKED ISSUE #89: Dev: Self Create Flow Contract
## Problem

Self Create flow should end up with the user signing, ideally, a single transaction. We will need a helper contract (the self create flow contract) so user signs this one transaction and we do the rest through cross-contract calls.

## Solution 

Is a helper contract that does these 4 steps in a single tx. The transaction should accomplish the following:

1. Deploy a DAO.
Create and configure a new Sputnik DAO subaccount using the Sputnik factory [create](https://github.com/near-daos/sputnik-dao-contract/blob/3d568f9517a8c7a6510786d978bb25b180501841/sputnikdao-factory2/src/lib.rs#L100) method. The create method on contract `sputnik-dao.near` should be used for this. It costs between 5 and 6 Near to deploy a DAO. The DAO contract will be a sub account of `sputnik-dao.near`. For example `new-treasury.sputnik-dao.near`.

2. Create a new account, for example `new-treasury.near`
This new account can be used to deploy the front end on but in order to store the components it needs a storage deposit. As a callback for making this new account we can do step 3.
 	
3. Pay for SocialDB storage + deploy BOS components
 These interactions are with the [social-db contract](https://github.com/NearSocial/social-db/blob/b4a023d234d0f24a2cf0d4b7998ea485f90f576e/contract/src/api.rs#L147)

4. On this new account we can deploy the web4 contract that is defined [here](https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/blob/5e18248872caed22e50d980d0156d8146c5efc97/web4/treasury-web4/src/lib.rs#L39) with BOS gateway.
Checkout [web4](https://web4.near.page/) for more info.

### PR DESCRIPTION
**Pull request by petersalomonsen**
A factory contract that

- Creates a new account using `near` linkdrop contract, the new account will be a direct account under `.near`
- Deploys the web4 contract to the new account
- Fetches latest widget content from the reference account you choose, for example `treasury-testing.near` ( `app` and `config` ) 
   - **NOTE**: When fetching widget content from `treasury-testing.near`, the config.jsx is hard-coded to use a specific sputnik-dao account. The reference config should be more generic, and automatically detect the account it is deployed to, and then also find the correct sputnik-dao account to use.
- Inserts widgets into `SocialDB` for the new account
- Creates a SputnikDAO instance

See [treasury-factory/tests/test_basics.rs](https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/blob/feat/treasury-factory-contract/treasury-factory/tests/test_basics.rs) for how to invoke it, and how it's being tested.

The treasury-factory contract is deployed to `treasury-factory.near`.

resolves #89

### CODE REVIEW

**Comment by petersalomonsen** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@petersalomonsen Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!


[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/115&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/115&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/petersalomonsen)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/petersalomonsen)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by frol — APPROVED** [review]
It looks great to me!

@race-of-sloths score 13

**Pull request merged**
